homework/insensitive-palindrom/palindrome.cpp

```diff
@@ -0,0 +1,11 @@
+#include "palindrome.hpp"
+#include <algorithm>
+#include <cctype>
+
+bool is_palindrome (std::string str) {
+    auto it = std::remove_if (str.begin(), str.end(), [](char letter){return !isalpha (letter);});
+    str.erase(it, str.end());
+    std::transform (str.begin(), str.end(), str.begin(), [](char letter){return std::toupper(letter);});
```

Comment:
You can get string as const & and use copy_if. Returned iterator will point to the end of the container (after copying only the letters) and use transform.

std::transform(begin.str(), std::copy_if(begin, end, begin));